ee/watchdog/controller_windows.go

```diff
@@ -180,224 +191,426 @@ func (wc *WatchdogController) ServiceEnabledChanged(enabled bool) {
 		return
 	}
 
-	var serviceManager *mgr.Mgr
-	var err error
+	if !enabled {
+		if err := RemoveWatchdogTask(wc.knapsack.Identifier()); err != nil {
+			wc.slogger.Log(ctx, slog.LevelWarn,
+				"encountered error removing watchdog task",
+				"err", err,
+			)
 
-	if err := backoff.WaitFor(func() error {
-		serviceManager, err = mgr.Connect()
-		if err != nil {
-			return fmt.Errorf("err connecting to service control manager: %w", err)
+			return
 		}
 
-		return nil
-	}, 5*time.Second, 500*time.Millisecond); err != nil {
+		wc.slogger.Log(ctx, slog.LevelInfo, "removed watchdog task")
+
+		return
+	}
+
+	// we're enabling the watchdog task- we can safely always reinstall our latest version here
+	if err := installWatchdogTask(wc.knapsack.Identifier(), wc.configFilePath); err != nil {
 		wc.slogger.Log(ctx, slog.LevelError,
-			"timed out connecting to service control manager",
+			"encountered error installing watchdog task",
 			"err", err,
 		)
+	}
 
-		return
+	wc.slogger.Log(ctx, slog.LevelInfo, "completed watchdog scheduled task installation")
+}
+
+func getExecutablePath(identifier string) (string, error) {
+	if strings.TrimSpace(identifier) == "" {
+		identifier = launcher.DefaultLauncherIdentifier
 	}
 
-	defer serviceManager.Disconnect()
+	binDirBase := fmt.Sprintf(`C:\Program Files\Kolide\Launcher-%s\bin`, identifier)
```

Comment:
Maybe we want to add this to `pkg/launcher/paths.go` so that we don't have two separate places where we define the bin directory?

Reply:
(Does `os.Executable()` work for this? Or not because of how we run launcher on Windows?)

Reply:
ah yeah it felt weird having this here, i'll look into somewhere better in paths, thanks! and yes the only issue with `os.Executable()` is that it could enter the scheduler database with an autoupdated exe path, and then have that path cleaned up after enough future autoupdates